flake.nix

```diff
@@ -97,98 +117,179 @@
 
         devShell = devenv.lib.mkShell {
           inherit inputs pkgs;
-          modules = with pkgs; [{
-            packages = [
-              elixir
-              elixir_ls
-              glibcLocales
-              node2nix
-              nodejs
-              prefetch-npm-deps
-              # for dashboard scripts
-              jq
-              psql
-              mosquitto
-              mosquitto_sub
-            ] ++ optional stdenv.isLinux inotify-tools
-              ++ optional stdenv.isDarwin terminal-notifier
-              ++ optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
-                CoreFoundation
-                CoreServices
-              ]);
-            enterShell = ''
-              export LOCALES="${cldr}/priv/cldr";
-              export PORT="4000"
-              export ENCRYPTION_KEY="your_secure_encryption_key_here"
-              export DATABASE_USER="teslamate"
-              export DATABASE_PASS="your_secure_password_here"
-              export DATABASE_NAME="teslamate"
-              export DATABASE_HOST="127.0.0.1"
-              export DATABASE_PORT="${toString postgres_port}"
-              export MQTT_HOST="127.0.0.1"
-              export MQTT_PORT="${toString mosquitto_port}"
-              export RELEASE_COOKIE="1234567890123456789"
-              export TZDATA_DIR="$PWD/tzdata"
-            '';
-            enterTest = ''
-              mix test
-            '';
-            processes.mqtt = {
-              exec =
-                "${pkgs.mosquitto}/bin/mosquitto -p ${toString mosquitto_port}";
-            };
-            process.process-compose = {
-              port = process_compose_port;
-              tui = true;
-            };
-            services.postgres = {
-              enable = true;
-              package = pkgs.postgresql_16;
-              listen_addresses = "127.0.0.1";
-              port = postgres_port;
-              initialDatabases = [{ name = "teslamate"; }];
-              initialScript = ''
-                CREATE USER teslamate with encrypted password 'your_secure_password_here';
-                GRANT ALL PRIVILEGES ON DATABASE teslamate TO teslamate;
-                ALTER USER teslamate WITH SUPERUSER;
+          modules = with pkgs; [
```

Comment:
This is probably my fault. But using `with` seems to be very much discouraged, the way it works is potentially confusing. Particularly on a large scope such as used here.

A number of people have written about the issues, and there are different options for replacing it. e.g. https://www.tweag.io/blog/2023-01-24-nix-with-with-nickel/

Reply:
This probably should be fixed after this PR is merged.